src/Runner/TaskHandler/Middleware/ParallelProcessingMiddleware.php

```diff
@@ -29,73 +26,35 @@ class ParallelProcessingMiddleware implements TaskHandlerMiddlewareInterface
      */
     private $poolFactory;
 
-    /**
-     * @var IOInterface
-     */
-    private $IO;
-
-    public function __construct(ParallelConfig $config, PoolFactory $poolFactory, IOInterface $IO)
+    public function __construct(ParallelConfig $config, PoolFactory $poolFactory)
     {
         $this->poolFactory = $poolFactory;
         $this->config = $config;
-        $this->IO = $IO;
     }
 
-    public function handle(TaskInterface $task, TaskRunnerContext $runnerContext, callable $next): Promise
-    {
+    public function handle(
+        TaskInterface $task,
+        TaskRunnerContext $runnerContext,
+        StopOnFailure $stopOnFailure,
+        callable $next
+    ): Future {
         if (!$this->config->isEnabled()) {
-            return $next($task, $runnerContext);
+            return async(static fn () => $next($task, $runnerContext, $stopOnFailure)->await());
         }
 
         $currentEnv = $_ENV;
+        $worker = $this->poolFactory->createShared();
+        $execution = $worker->submit(
```

Comment:
The error handling of the new worker system seems a lot better.
Do I need to take care of some other amp related exceptions?

(see old implementation -> There I needed to wrap a lot of error handling myself)